tests/c_languages/CPPCheckBearTest.py

```diff
@@ -30,19 +32,46 @@ def get_results(self, files_to_check):
         self.uut = CPPCheckBear(self.file_dict,
                                 self.section,
                                 self.queue)
-        return list(self.uut.run_bear_from_section([], {}))
 
-    def test_results_complete(self):
+        result = self.uut.run_bear_from_section([], {})
+        if result is not None:
+            return list(result)
+        return
+
+    def test_results_complete_language_cpp(self):
         self.section.append(Setting('enable', 'unusedFunction'))
+        self.section.append(Setting('language', 'c++'))
         results = self.get_results(self.test_files)
         messages = [result.message for result in results]
-        self.assertEqual(len(messages), 2)
+        self.assertEqual(len(messages), 4)
         self.assertRegex(messages[0], 'Array .+ out of bounds')
-        self.assertRegex(messages[1], "function 'f1' .+ never used")
+        self.assertRegex(messages[1], 'When .+ out of bounds')
+        self.assertRegex(messages[2], "function 'f1' .+ never used")
+        self.assertRegex(messages[3], "function 'foo' .+ never used")
 
-    def test_no_enable_entered(self):
+    def test_language_c(self):
+        self.section.append(Setting('language', 'c'))
         results = self.get_results(self.test_files)
         messages = [result.message for result in results]
         self.assertEqual(len(messages), 1)
```

Comment:
Test were reporting a different number. Please check that!

Reply:
This test is getting passed on my local machine. I am not sure what else it is expecting?

Reply:
May the messages count is greater than 3, due to above reason. Just check with that first.
btw i reported this change request because https://ci.appveyor.com/project/coala/coala-bears/builds/26862272/job/81isokrqkuiupbf1#L1446 this was logging that there are 3 messages